controllers/apim/utils.go

```diff
@@ -0,0 +1,81 @@
+package apim
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kuadrant/limitador-operator/api/v1alpha1"
+	istio "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	istiosecurityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// gatewayLabels fetches labels of an Istio gateway identified using the given ObjectKey.
+func gatewayLabels(ctx context.Context, client client.Client, gwKey client.ObjectKey) map[string]string {
+	gateway := &istio.Gateway{}
+	if err := client.Get(ctx, gwKey, gateway); err != nil {
+		return map[string]string{}
+	}
+	return gateway.Labels
```

Comment:
It is not the labels of the gateway object. It's about the labels of the workload pod (i.e. the actual gateway process). 

the `gateway.spec.selector` should be returned

Reply:
Fixed